app/models/internal/reports/ReportFormat.java

```diff
@@ -40,30 +40,30 @@
      *
      * @param <T> the return type of the visitor.
      */
-    interface Visitor<T> {
+    abstract class Visitor<T> {
         /**
          * Visit an {@code PdfReportFormat}.
          *
          * @param pdfReportFormat The format to visit.
          * @return The result of applying the visitor.
          */
-        T visit(PdfReportFormat pdfReportFormat);
+        public abstract T visitPdf(PdfReportFormat pdfReportFormat);
 
         /**
          * Visit an {@code HtmlReportFormat}.
          *
          * @param htmlReportFormat The format to visit.
          * @return The result of applying the visitor.
          */
-        T visit(HtmlReportFormat htmlReportFormat);
+        public abstract T visitHtml(HtmlReportFormat htmlReportFormat);
 
         /**
          * Convenience method equivalent to {@code format.accept(this) }.
          *
          * @param format The format to visit.
          * @return The result of applying the visitor
          */
-        default T visit(ReportFormat format) {
+        public final T visit(final ReportFormat format) {
             return format.accept(this);
```

Comment:
Sorry, is there a reason the called does not call `accept` on the object and pass the visitor?